### ISSUE TITLE
Support the Clojure nREPL CIDER debugger

### ISSUE THREAD

**Issue by Olical** [Olical/conjure#182]
Does what it says on the tin. Let's have debugger support, should probably just a pass through nREPL and support the newer features, so probably involves doing #137 too! Maybe #139 as well.

**Comment by Olical** [comment]
https://github.com/puremourning/vimspector seems like a promising frontend for the debugger! Implementing the UI is kinda out of scope I think, unless I can think of a good text based log buffer interface... so I think this could be a great shortcut to a great debugger experience.

**Comment by Olical** [comment]
Thank you to @dharrigan for prompting my interest in this 😄

**Comment by glyh** [comment]
I think debugging functionalities is beyond the scope of this plugin. Maybe we can add good interaction with vimspector though.

**Comment by Olical** [comment]
Agreed, still a sizeable chunk of work to design that interface for each client to implement though. I also worry only very few clients transports / implementations will be powerful enough to allow it.

**Comment by awwalker** [comment]
@Olical or @glyh do you know if there are any docs on how the nREPL handles debug request/responses? My thinking is that there would need to be a shim between those messages and the [DAP](https://microsoft.github.io/debug-adapter-protocol/specification) specification so that plugins like [nvim-dap](https://github.com/mfussenegger/nvim-dap) can handle the ui and client stuff.

**Comment by Olical** [comment]
I'm not sure of any documentation right now although I'm away on holiday at
the moment so haven't had time to look it up. I'm sure some exists! I will
be digging into this as part of my Clojurist Together funding in April
however ☺️

**Comment by awwalker** [comment]
Please let me know if I can be of any assistance, I'm happy to start looking for docs, etc. now! 

I'm just starting my Clojure journey but have some experience with DAP and would love to make debugging a first class workflow for conjure users. Enjoy the rest of your holiday and hopefully hear from you again in April.

**Comment by glyh** [comment]
I guess you need to look through [nREPL](https://nrepl.org/nrepl/index.html) and [CIDER](https://docs.cider.mx/cider/index.html) websites? 

And maybe the best way to understand nREPL is to read through some nREPL servers' source codes.

**Comment by Olical** [comment]
We now have _some_ CIDER debugger support with plans for a standalone tool that will allow you to connect _any_ DAP client (in any editor, including Neovim) to a CIDER nREPL to drive it's debugger tolling with clever GUIs.

See here for information about Conjure's built in debugger tooling but I hope to supersede it with the standalone tooling https://github.com/Olical/conjure/wiki/Clojure-nREPL-CIDER-debugger
